code/acting/src/acting/trajectory_interpolation.py

```diff
@@ -0,0 +1,132 @@
+"""A module for interpolating routes"""
+
+from math import dist as euclid_dist, floor, sqrt, sin, cos
+from typing import List, Tuple
+
+
+def points_to_vector(p_1: Tuple[float, float],
+                     p_2: Tuple[float, float]) -> Tuple[float, float]:
+    """Create the vector starting at p1 and ending at p2"""
+    return p_2[0] - p_1[0], p_2[1] - p_1[1]
+
+
+def vector_len(vec: Tuple[float, float]) -> float:
+    """Compute the given vector's length"""
+    return sqrt(vec[0]**2 + vec[1]**2)
+
+
+def add_vector(v_1: Tuple[float, float],
+               v_2: Tuple[float, float]) -> Tuple[float, float]:
+    """Add the given vectors"""
+    return v_1[0] + v_2[0], v_1[1] + v_2[1]
+
+
+def scale_vector(vector: Tuple[float, float],
+                 new_len: float) -> Tuple[float, float]:
+    """Amplify the length of the given vector"""
+    old_len = vector_len(vector)
+    if old_len == 0:
+        return (0, 0)
+    scaled_vector = (vector[0] * new_len / old_len,
+                     vector[1] * new_len / old_len)
+    return scaled_vector
+
+
+def rotate_vector(vector: Tuple[float, float],
+                  angle_rad: float) -> Tuple[float, float]:
+    """Rotate the given vector by an angle"""
+    return (cos(angle_rad) * vector[0] - sin(angle_rad) * vector[1],
+            sin(angle_rad) * vector[0] + cos(angle_rad) * vector[1])
+
+
+def linear_interpolation(start: Tuple[float, float], end: Tuple[float, float],
+                         interval_m: float) -> List[Tuple[float, float]]:
+    """Interpolate linearly between the given start / end point
+    by putting points according to the interval specified."""
+
+    distance = euclid_dist(start, end)
+    vector = (end[0] - start[0], end[1] - start[1])
```

Comment:
could you explain, why you didn't use your points_to_vector_function which you explictly wrote for that purpose?

Reply:
thanks, fixed in 93021b8